include/miral/miral/window_manager_tools.h

```diff
@@ -39,229 +39,273 @@ struct ApplicationInfo;
-     * @return          the metadata
-     */
+    /// Retrieve info about a window from the \p surface.
+    ///
+    /// \param surface the surface
+    /// \returns info about the window, or the default-constructed window info if the \p surface is invalid
+    /// \sa miral::WindowInfo - info about the window
     auto info_for(std::weak_ptr<mir::scene::Surface> const& surface) const -> WindowInfo&;
 
-    /** retrieve metadata for a window
-     *
-     * @param window    the window
-     * @return          the metadata
-     */
+    /// Retrieve info about a window.
+    ///
+    /// \param window the window
+    /// \returns info about the window, or the default-constructed window info if the \p surface is invalid
+    /// \throws std::out_of_range when the window is invalid
+    /// \sa miral::WindowInfo - info about the window
     auto info_for(Window const& window) const -> WindowInfo&;
 
-    /** retrieve metadata for a persistent surface id
-     *
-     * @param id        the persistent surface id
-     * @return          the metadata
-     * @throw           invalid_argument or runtime_error if the id is badly formatted/doesn't identify a current window
-     * \deprecated      'Persistent' surface IDs were part of mirclient API
-     */
+    /// Retrieve info for a persistent surface id.
+    ///
+    /// \param id the persistent surface id
+    /// \returns the info
+    /// \throws invalid_argument or runtime_error if the id is badly formatted/doesn't identify a current window
+    /// \deprecated 'Persistent' surface IDs were part of mirclient API
     [[deprecated("'Persistent' surface IDs were part of mirclient API")]]
     auto info_for_window_id(std::string const& id) const -> WindowInfo&;
 
-    /** retrieve the persistent surface id for a window
-     *
-     * @param window    the window
-     * @return          the persistent surface id
-     * \deprecated      'Persistent' surface IDs were part of mirclient API
-     */
+    /// Retrieve the persistent surface id for a window
+    ///
+    /// \param window the window
+    /// \returns the persistent surface id
+    /// \deprecated 'Persistent' surface IDs were part of mirclient API
     [[deprecated("'Persistent' surface IDs were part of mirclient API")]]
     auto id_for_window(Window const& window) const -> std::string;
 
-    /// Send close request to the window
+    /// Send a close request to the window.
+    ///
+    /// \param window the window
     void ask_client_to_close(Window const& window);
 
-    /// retrieve the active window
+    /// Retrieve the active window.
+    ///
+    /// If no window is active, then this will be the default-constructed window.
+    ///
+    /// \returns the active window
     auto active_window() const -> Window;
 
-    /** select a new active window based on the hint
-     *
-     * @param hint  the hint
-     * @return      the new active window
-     */
+    /// Select a new active window based on the \p hint.
```

Comment:
```suggestion
    /// Select a new active window based on the \p hint. 
    /// Not all windows can become active - either because of their type
    /// or other state. But there will typically be an associated parent or
    /// child window that can become active.
```